### PR TITLE
Implementar Query Suggestions - Algolia

### LINKED ISSUE #293: Implementar Query Suggestions - Algolia
## Feature Description

https://www.algolia.com/doc/guides/getting-insights-and-analytics/leveraging-analytics-data/query-suggestions/

A ideia aqui é basicamente usar o recurso Query Suggestions do Algolia e no lugar de fazer um "Autocomplete" a busca no `SearchForm` vai retornar sugestões (exemplo https://preview.algolia.com/delayed-hits-demo/?q=computer%20tablet&idx=instant_search&p=0)

---------------

## Acceptance criteria

* Ao buscar, os resultados retornados devem ser sugestões e não tecnologias em si.
* No lugar de exibir a imagem e descrição, alterar renderizar simplesmente a sugestão como no link acima.

## Implementation Brief

* Criar um novo component `SearchSuggestions` para usar no lugar do `Autocomplete`

### PR DESCRIPTION
**Pull request by luizeboli**
## Summary

Resolves #293

https://www.loom.com/share/5b28f579be2847d4b1fd26441d3f3a8b

## Relevant technical choices

* Created an API method to configure the query suggestions index
* Changed `indexName` in web  to the query suggestions one, so all searches points to query suggestion.

## QA Steps

**Watch video on above link**

1. Navigate to `packages/api` folder
2. Exec `npm run algolia:index` script
3. Access your Algolia dashboard
4. Assert that query suggestions index has been created in Query Suggestions menu
5. Access app homepage
6. Try to search for a term in the search bar (**actually only category will work, but after merging PR #583, other terms should work too**)
7. It should return terms suggestions instead of technologies.

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code passes the linting.
- [x] My code has proper inline documentation.
- [x] I have manually tested this PR.

### CODE REVIEW

File: packages/api/app/Commands/AlgoliaIndex.js [hunk 1]
```diff
@@ -58,13 +59,16 @@ class AlgoliaIndex extends Command {
...
+				);
+
 				const tec = {
 					...item,
-					category: item.terms.find(
-						(term) =>
-							term.taxonomy.taxonomy === CATEGORY_TAXONOMY_SLUG &&
-							term.parent_id === null,
-					).term,
+					category: categoryTerm ? categoryTerm.term : 'Não definida',
```
Comment (luizeboli) [inline packages/api/app/Commands/AlgoliaIndex.js:71]: Refactoring to safeguard this code agains't null categories

File: packages/web/public/static/locales/en/search.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "filter": "Filter",
   "filterOnlyPublic": "Only public technologies",
-  "foundTerms": "{{nbHits}} result(s) found for the search term \"{{termQuery}}\" in {{timeSpentMS}}ms",
+  "foundTerms": "{{nbHits}} result(s) found in {{timeSpentMS}}ms",
```
Comment (luizeboli) [inline packages/web/public/static/locales/en/search.json:4]: Had to remove search term from stats due to react-autosuggest. It changes search term while navigating through suggestions.

File: packages/web/config.js [hunk 3]
```diff
@@ -33,6 +33,7 @@ if (typeof window === 'undefined' || process.env.APP_ENV === 'testing') {
 		ALGOLIA_SEARCH_KEY: process.env.ALGOLIA_SEARCH_KEY,
 		ALGOLIA_APPLICATION_ID: process.env.ALGOLIA_APPLICATION_ID,
 		ALGOLIA_INDEX: process.env.ALGOLIA_INDEX,
+		ALGOLIA_QUERY_SUGGESTIONS_INDEX: process.env.ALGOLIA_QUERY_SUGGESTIONS_INDEX,
```
Comment (lcnogueira) [inline packages/web/config.js:36]: A quick question: shouldn't we include this for the other environments (`testing`, `staging`, `production`) as well?

Also, please, include it in the `.env.example` file 👍

**Review by lcnogueira — DISMISSED** [review]
Great job as usual, Felício 😃

**Review by calaca — DISMISSED** [review]
All good! :clap:

**Comment by luizeboli** [comment]
@lcnogueira ty, added this index name to other envs and to env.example 👍

**Comment by nichollasrennah** [comment]
Já foi resolvido tudo @luizeboli ? sabe informar se dá pra concluir hoje?

**Comment by luizeboli** [comment]
> Já foi resolvido tudo @luizeboli ? sabe informar se dá pra concluir hoje?

Já foi resolvido tudo. Só aguardando aprovação. Não pode esquecer da outra pr que complementa essa. Da api. Está na mesma situação

File: packages/web/.env.example [hunk 4]
```diff
@@ -2,5 +2,6 @@ NODE_ENV=development
 ALGOLIA_SEARCH_KEY=
 ALGOLIA_APPLICATION_ID=
 ALGOLIA_INDEX="searchable_data"
+ALGOLIA_QUERY_SUGGESTIONS_INDEX="searchable_data_query_suggestions",
```
Comment (calaca) [inline packages/web/.env.example:5]: :100:

**Review by calaca — APPROVED** [review]
Good job @luizeboli !

**Review by mateus4k — APPROVED** [review]

**Pull request merged**